core/common/src/main/java/alluxio/collections/IndexedSet.java

```diff
@@ -274,10 +268,8 @@ public boolean contains(FieldIndex<T> index, Object value) {
    * @return the set of objects or an empty set if no such object exists
    */
   public Set<T> getByField(FieldIndex<T> index, Object value) {
-    synchronized (mLock) {
-      Set<T> set = getByFieldInternal(index, value);
-      return set == null ? new HashSet<T>() : set;
-    }
+    Set<T> set = getByFieldInternal(index, value);
+    return set == null ? new HashSet<T>() : set;
```

Comment:
This method doesn't seem to be used, but if we plan to keep it, perhaps make it an unmodifiable version?